psychopy/visual/dot.py

```diff
@@ -482,19 +482,20 @@ def _update_dotsXY(self):
         if sum(dead):
             self._verticesBase[dead, :] = self._newDotsXY(sum(dead))
             #self._verticesBase[dead, :] = -self._verticesBase[dead,:]
-            
+
         # Reposition any dots that have gone out of bounds. Net effect is to place dot one step inside the boundary on the other side of the aperture.
         if sum(outofbounds):
+            self._verticesBase[outofbounds, :] = self._newDotsXY(sum(outofbounds))
             #wind the dots back one step and store as tempary values
-            if self.noiseDots == 'position':
```

Comment:
@schofiaj For now I've turned off your wrapping fix and reverted to recreating dots with a new random position, due to the issue with dots appearing outside the aperture. Your other fixes remain in place though.